cbor/src/main/java/com/fasterxml/jackson/dataformat/cbor/CBORParser.java

```diff
@@ -245,6 +322,35 @@ private Feature(boolean defaultState) {
      */
     protected int _typeByte;
 
+    /**
+     * The depth of nested objects.
+     */
+    protected int _nestedDepth = 0;
+
+    /**
+     * Type to keep track of a list of string references. A depth is stored to know when to pop the
+     * references off the stack for nested namespaces.
+     */
+    protected static final class StringRefList
+    {
+        public StringRefList(int depth) {
+            this.depth = depth;
+        }
+
+        public ArrayList<Object> stringRefs = new ArrayList<>();
+        public int depth;
+    }
+
+    /**
+     * Stack of text and binary string references.
+     */
+    protected Stack<StringRefList> _stringRefs = new Stack<>();
```

Comment:
Could probably create sub-class that has helper methods, if that simplifies calling code (will add a note where this'd help)